### PR TITLE
feat(ha-cluster-agent): add ha-cluster-agent rpc

### PR DESCRIPTION
**Pull request by mynktl**
Signed-off-by: Mayank <mayank.patel@datacore.com>

### CODE REVIEW

File: protobuf/ha_cluster_agent.proto [hunk 1]
```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
```
Comment (mtzaurus) [inline protobuf/ha_cluster_agent.proto:1]: Shall we consider this API also versioned and put it inside v1 scope ?

Comment (tiagolobocastro) [reply protobuf/ha_cluster_agent.proto:1]: I think so but will this api be used by the data plane? If this is for the control-plane only it should go here: https://github.com/openebs/mayastor-control-plane/tree/develop/control-plane/grpc/proto/v1

Comment (mynktl) [reply protobuf/ha_cluster_agent.proto:1]: I have added this changes to control-plane PR.

**Comment by mynktl** [comment]
Closing this PR since api will be used in control-plane only. Relevant PR : https://github.com/openebs/mayastor-control-plane/pull/277

**Pull request closed without merging**